src/checks/duplicated_keys.rs

```diff
@@ -0,0 +1,125 @@
+use std::collections::HashMap;
+
+use crate::checks::{GlobalCheck, Warning};
+use crate::extract_key;
+use crate::LineEntry;
+
+#[derive(Debug)]
+pub(crate) struct DuplicatedKeysChecker {
+    template: String,
+}
+
+impl Default for DuplicatedKeysChecker {
+    fn default() -> Self {
+        Self {
+            template: String::from("The {} key is duplicated"),
+        }
+    }
+}
+
+impl GlobalCheck for DuplicatedKeysChecker {
+    fn run(&self, lines: Vec<LineEntry>) -> Vec<Warning> {
```

Comment:
I did a bit refactoring yesterday - https://github.com/mgrachev/dotenv-linter/pull/52

So you can implement this check easier:
```rust
pub(crate) struct DuplicatedKeysChecker {
    template: String,
    keys: Vec<String>,
}

impl Check for DuplicatedKeysChecker {
    fn run(&mut self, line: LineEntry) -> Option<Warning> {
        // get a key

        // check if the key exists in the vector of keys
        // If exists - return Some

        // Push the key to the vector and return None
    }
}
```

Reply:
Thank you, that is way better :)